internal/core/pipeline.go

```diff
@@ -0,0 +1,65 @@
+package core
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/config"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+// Pipeline represents a single pipeline and exposes all of the operations
+// that can be performed on a pipeline.
+type Pipeline struct {
```

Comment:
Oh okay I actually like this model! (Making it a thing so you can do `core.pipeline.sync` etc.)